resolwe_runtime_utils.py

```diff
@@ -275,12 +288,177 @@ def checkrc(rc, *args):
     return json.dumps(ret)
 
 
+CHUNK_SIZE = 10000000  # 10 Mbytes
+
+
+class ImportedFormat:
+    """Import destination file format."""
+
+    EXTRACTED = 'extracted'
+    COMPRESSED = 'compressed'
+    BOTH = 'both'
+
+
+def import_file(src, file_name, imported_format=ImportedFormat.BOTH, max_progress=None):
+    """Import file to working directory.
+
+    :param src: Source file path or URL
+    :param file_name: Source file name
+    :param imported_format: Import file format (extracted, compressed or both)
+    :param max_progress: Progress value after import
+    :return: Destination file path (if extracted and compressed, extracted path given)
+    """
+    print("Importing and compressing {}...".format(file_name))
+
+    def importGz():
+        """Import gzipped file.
+
+        The file_name must have .gz extension.
+        """
+        if imported_format != ImportedFormat.EXTRACTED:  # Compressed file required
+            with gzip.open(src, 'rb') as f:
+                try:
+                    while f.read(CHUNK_SIZE) != b'':
+                        pass
+                except Exception:
+                    raise ValueError("invalid gzip file format")
```

Comment:
Consider including the file name. Don't we want to `print(error(...))` here as well?

Reply:
Refactored to print